src/cpp/src/tokenizer.cpp

```diff
@@ -86,11 +87,16 @@ class Tokenizer::TokenizerImpl {
 
     std::string m_chat_template = {};
 
-    void set_state_if_necessary(CircularBufferQueueElementGuard<ov::InferRequest>& infer_request_guard, bool add_special_tokens) {
+    void set_state_if_necessary(CircularBufferQueueElementGuard<ov::InferRequest>& infer_request_guard, const ov::AnyMap& params) {
+        bool add_special_tokens_flag = true;
+        bool skip_special_tokens_flag = false;
+        ov::genai::utils::read_anymap_param(params, add_special_tokens.name(), add_special_tokens_flag);
+        ov::genai::utils::read_anymap_param(params, skip_special_tokens.name(), skip_special_tokens_flag);
```

Comment:
does it mean that if I run:
```c++
tok.encode("ccc", ov::genai::skip_special_tokens(true))
```
it will also allowed? 
there is distinguishing between tokenization and de-tokenization parameters.

Reply:
Yes, tokenization and detokenization parameters differ. This is done to be aligned with HF